### PR TITLE
Adds rollback test

### PR DESCRIPTION
**Pull request by jeremy-then**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2510,6 +2510,45 @@ void voteFederationChange_addFederatorMultiKey_returnsExpectedPendingFederationS
...
+        @Test
+        void voteFederationChange_rollback_returnsSuccessResponseCode() {
+
+            // Arrange
+
+            BtcECKey expectedBtcECKey = new BtcECKey();
+            ECKey expectedRskKey = new ECKey();
+            ECKey expectedMstKey = new ECKey();
+
+            Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2522]: Use meaningful names. e.g. `firstAuthorizedTx`

**Review by nathanieliov — APPROVED** [review]
Just a naming suggestion

**Comment by marcos-iov** [comment]
Seems to be some rebasing issues here @jeremy-then

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -2509,6 +2510,45 @@ void voteFederationChange_addFederatorMultiKey_returnsSuccessfulResponseCode() {
...
+
+            ABICallSpec rollbackAbiCallSpec = new ABICallSpec(FederationChangeFunction.ROLLBACK.getKey(), new byte[][]{
+                expectedFederatorBtcECKey.getPubKey(),
+                expectedFederatorRskKey.getPubKey(),
+                expectedFederatorMstKey.getPubKey()
+            });
+
+            // Act
+
+            // Voting add new fed with m of n authorizers
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2539]: Wrong comment, anyway I would first vote for at least 1 member before doing the rollback. To make it more complete

Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 3]
```diff
@@ -2528,23 +2526,32 @@ void voteFederationChange_rollback_returnsSuccessResponseCode() {
             federationSupport.voteFederationChange(firstAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
             federationSupport.voteFederationChange(secondAuthorizedTx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
 
-            ABICallSpec rollbackAbiCallSpec = new ABICallSpec(FederationChangeFunction.ROLLBACK.getKey(), new byte[][]{
-                expectedFederatorBtcECKey.getPubKey(),
-                expectedFederatorRskKey.getPubKey(),
-                expectedFederatorMstKey.getPubKey()
-            });
+            ABICallSpec addFederationMemberAbiCallSpec = new ABICallSpec(FederationChangeFunction.ADD.getKey(), new byte[][]{expectedFederatorBtcECKey.getPubKey()});
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2529]: See the comments in other PRs, this is disabled at the moment

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 4]
```diff
@@ -2509,6 +2510,52 @@ void voteFederationChange_addFederatorMultiKey_returnsSuccessfulResponseCode() {
...
+
+            ABICallSpec addFederationMemberAbiCallSpec = new ABICallSpec(FederationChangeFunction.ADD.getKey(), new byte[][]{expectedFederatorBtcECKey.getPubKey()});
+
+            // Act
+
+            // Voting add new fed with m of n authorizers to have at least one federation member before roll back
+            federationSupport.voteFederationChange(firstAuthorizedTx, addFederationMemberAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(secondAuthorizedTx, addFederationMemberAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            ABICallSpec rollbackAbiCallSpec = new ABICallSpec(FederationChangeFunction.ROLLBACK.getKey(), new byte[][]{});
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2537]: shouldn't this be on Arrange part?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 5]
```diff
@@ -2509,6 +2510,52 @@ void voteFederationChange_addFederatorMultiKey_returnsSuccessfulResponseCode() {
...
+
+            // Act
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2539]: why are there two Act parts?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 4 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2541]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 5 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2543]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**